src/argus/incident/signals.py

```diff
@@ -36,3 +38,38 @@ def send_notification(sender, instance: Event, *args, **kwargs):
 def delete_associated_event(sender, instance: Acknowledgement, *args, **kwargs):
     if hasattr(instance, "event") and instance.event:
         instance.event.delete()
+
+
+def detect_changes(sender, instance: Incident, raw, *args, **kwargs):
+    if raw or not instance.id or not instance.end_time:
+        return
+    instance._changed = {}
+    previous = Incident.objects.get(id=instance.id)
+    if previous.level != instance.level:
+        instance._changed["level"] = previous.level
+    if previous.details_url != instance.details_url:
+        instance._changed["details_url"] = previous.details_url
+    if previous.ticket_url != instance.ticket_url:
+        instance._changed["ticket_url"] = previous.ticket_url
+
+
+def create_change_events(sender, instance: Incident, created, raw, *args, **kwargs):
+    if instance and getattr(instance, "_changed", None):
+        if "level" in instance._changed.keys():
+            old_level = instance._changed["level"]
+            description = f"Change: level {old_level} → {instance.level}"
+            ChangeEvent.objects.create(
+                incident=instance, actor=instance.user, timestamp=timezone.now(), description=description
+            )
+        if "details_url" in instance._changed.keys():
+            old_details_url = instance._changed["details_url"]
+            description = f"Change: details_url {old_details_url} → {instance.details_url}"
+            ChangeEvent.objects.create(
+                incident=instance, actor=instance.user, timestamp=timezone.now(), description=description
+            )
+        if "ticket_url" in instance._changed.keys():
+            old_ticket_url = instance._changed["ticket_url"]
+            description = f"Change: ticket_url {old_ticket_url} → {instance.ticket_url}"
+            ChangeEvent.objects.create(
+                incident=instance, actor=instance.source.user, timestamp=timezone.now(), description=description
+            )
```

Comment:
IMHO,

One time is one time.
Two times is ok.
Three times is a pattern :)

```suggestion
def create_change_events(sender, instance: Incident, created, raw, *args, **kwargs):
    if not instance or not getattr(instance, "_changed", None):
        return

    for attr in ("level", "details_url", "ticket_url"):
        if attr not in  instance._changed.keys():
            continue

        old_value = instance._changed[attr]
        new_value = getattr(instance, attr)
        description = f"Change: level {old_value} → {new_value}"
        ChangeEvent.objects.create(
            incident=instance, actor=instance.user, timestamp=timezone.now(), description=description
        )
```